tests/basic/bug68986.phpt

```diff
@@ -0,0 +1,23 @@
+--TEST--
+Bug #68986 (pointer returned by php_stream_fopen_temporary_file not validated in memory.c)
+--INI--
+default_charset=UTF-8
+--FILE--
+<?php
+if (substr(PHP_OS, 0, 3) == 'WIN') {
+	die('skip.. only for unix');
+}
+
+$tempDir = getenv("TMPDIR");
+mkdir($tempDir . '/php68986');
+system("chmod 444 " . $tempDir . '/php68986');
+putenv("TMPDIR=" . $tempDir . '/php68986');
+
+$fp = fopen('php://temp', 'r+');
+$data = implode('', array_fill(0, (1024 * 1024 * 2), 'A'));
+var_dump(fwrite($fp, $data)); 
+fclose($fp);
+rmdir($tempDir . '/php68986');
```

Comment:
I set 444 to create a ready only temp dir, where earlier a segfault was
occurring, my test was to cover that segfault no longer occurs,

Strangely though when writing 1024 × 1024 × 2 bytes to a php://temp with
read only condition fwrite reports a slight lower number than 2mb has been
written

So maybe there is another place we need to fix

Sent via mobie
On Feb 5, 2015 2:58 PM, "Xinchen Hui" <notifications@github.com> wrote:

> In tests/basic/bug68986.phpt
> <https://github.com/php/php-src/pull/1051#discussion_r24146247>:
>
> > +--FILE--
> > +<?php
> > +if (substr(PHP_OS, 0, 3) == 'WIN') {
> > +	die('skip.. only for unix');
> > +}
> > +
> > +$tempDir = getenv("TMPDIR");
> > +mkdir($tempDir . '/php68986');
> > +system("chmod 444 " . $tempDir . '/php68986');
> > +putenv("TMPDIR=" . $tempDir . '/php68986');
> > +
> > +$fp = fopen('php://temp', 'r+');
> > +$data = implode('', array_fill(0, (1024 * 1024 * 2), 'A'));
> > +var_dump(fwrite($fp, $data));
> > +fclose($fp);
> > +rmdir($tempDir . '/php68986');
>
> I doubt this will success, since you have chmod 444
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/php/php-src/pull/1051/files#r24146247>.
>